lib/intervention/client.rb

```diff
@@ -1,34 +1,47 @@
 module Intervention
   class Client < EventMachine::Connection
-    attr_reader :server, :parser
+    attr_reader :server, :parser, :enable_tls
+
+    def initialize(config = nil)
+      @enable_tls = !config.nil?
```

Comment:
So, this client class is used to both servers (http and https). In case of the http server there is no need to pass any extra config. But in case of the https server there is the need to pass extra config for the class, and I am using that fact to config the variables. 

But you are right. It isn't clear. I will change to:

```
def initialize(config = nil)
      activate_https_if_needed(config)

      super
    end
    
    def activate_https_if_needed(config)
      return unless config

      @enable_tls = true
      @config     = config
    end
```


Do you agree?